File: RailgunVB/Core/Utilities/ServerCount.vb
Imports System.Timers
Imports Discord
Imports Discord.WebSocket
Imports RailgunVB.Core.Configuration

Namespace Core.Utilities
    
    Public Class ServerCount
        Private ReadOnly _config As MasterConfig
        Private ReadOnly _client As DiscordShardedClient
        
        Private WithEvents _timer As New Timer(TimeSpan.FromMinutes(30).TotalMilliseconds)
        
        Public Property PreviousGuildCount As Integer = 0

        Public Sub New(config As MasterConfig, client As DiscordShardedClient)
            _config = config
            _client = client
            
            _timer.AutoReset = True
            _timer.Start()
        End Sub
        
        Private Async Sub UpdateServerCountAsync() Handles _timer.Elapsed
            If PreviousGuildCount = _client.Guilds.Count Then Return
            
            PreviousGuildCount = _client.Guilds.Count
            
            Await _client.SetGameAsync($"{_config.DiscordConfig.Prefix}help || {PreviousGuildCount} Servers!",
                type := ActivityType.Watching)
        End Sub
        
    End Class
    
End NameSpace

File: Railgun/Core/Utilities/ServerCount.cs
using System;
using System.Threading.Tasks;
using System.Timers;
using Discord;
using Discord.WebSocket;
using Railgun.Core.Configuration;

namespace Railgun.Core.Utilities
{
    public class ServerCount
    {
        private readonly MasterConfig _config;
        private readonly DiscordShardedClient _client;
        private Timer _timer = new Timer(TimeSpan.FromMinutes(30).TotalMilliseconds);

        public int PreviousGuildCount { get; set; } = 0;

        public ServerCount(MasterConfig config, DiscordShardedClient client) {
            _config = config;
            _client = client;

            _timer.AutoReset = true;
            _timer.Elapsed += async (sender, args) => await UpdateServerCountAsync();
            _timer.Start();
        }

        private async Task UpdateServerCountAsync() {
            if (PreviousGuildCount == _client.Guilds.Count) return;

            PreviousGuildCount = _client.Guilds.Count;

            await _client.SetGameAsync($"{_config.DiscordConfig.Prefix}help || {PreviousGuildCount} Servers!", type:ActivityType.Watching);
        }
    }
}